modules/core/server/services/email.server.service.js

```diff
@@ -0,0 +1,219 @@
+      campaign: 'reset-password'
+    })
+  });
+  exports.renderEmailAndSend('reset-password', params, callback);
+};
+
+exports.sendResetPasswordConfirm = function(user, callback) {
+  var params = exports.addEmailBaseTemplateParams({
+    subject: 'Your password has been changed',
+    name: user.displayName,
+    email: user.email,
+    utmCampaign: 'reset-password-confirm'
+  });
+  exports.renderEmailAndSend('reset-password-confirm', params, callback);
+};
+
+exports.sendEmailConfirmation = function(user, callback) {
+
+  var urlConfirm = url + '/confirm-email/' + user.emailToken + '?signup',
+      utmCampaign = 'confirm-email';
+
+  var params = exports.addEmailBaseTemplateParams({
+    subject: 'Confirm Email',
+    name: user.displayName,
+    email: user.email,
+    urlConfirmPlainText: urlConfirm,
+    urlConfirm: analyticsHandler.appendUTMParams(urlConfirm, {
+      source: 'transactional-email',
+      medium: 'email',
+      campaign: utmCampaign
+    }),
+    utmCampaign: utmCampaign
+  });
+
+  exports.renderEmailAndSend('email-confirmation', params, callback);
+};
+
+/**
+ * Add several parameters to be used to render transactional emails
+ * These variables are used by email base template:
+ * `modules/core/server/views/email-templates/email.server.view.html`
+ *
+ * @param {Object[]} params - Parameters used for rendering emails
+ * @returns {Object[]} - Returns object with supportUrl, footerUrl and headerUrl parameters.
+ */
+exports.addEmailBaseTemplateParams = function(params) {
+  if (params === null || typeof params !== 'object') {
+    console.error('addEmailBaseTemplateParams: requires param to be Object. No URL parameters added.');
+    return {};
+  }
+
+  var baseUrl = (config.https ? 'https' : 'http') + '://' + config.domain;
+
+  params.urlSupportPlainText = baseUrl + '/support';
+  params.footerUrlPlainText = baseUrl;
+
+  params.headerUrl = analyticsHandler.appendUTMParams(baseUrl, {
+    source: 'transactional-email',
+    medium: 'email',
+    campaign: params.utmCampaign || 'transactional-email',
+    content: 'email-header'
+  });
+
+  params.footerUrl = analyticsHandler.appendUTMParams(baseUrl, {
+    source: 'transactional-email',
+    medium: 'email',
+    campaign: params.utmCampaign || 'transactional-email',
+    content: 'email-footer'
+  });
+
+  params.supportUrl = analyticsHandler.appendUTMParams(params.urlSupportPlainText, {
+    source: 'transactional-email',
+    medium: 'email',
+    campaign: params.utmCampaign || 'transactional-email',
+    content: 'email-support'
+  });
+
+  return params;
+};
+
+exports.renderEmail = function(templateName, params, callback) {
+
+  var htmlTemplatePath = path.join(htmlTemplateDir, templateName + '.server.view.html');
+  var textTemplatePath = path.join(textTemplateDir, templateName + '.server.view.html');
+
+  // Rendering in parallel leads to an error. maybe because
+  // swig is unmaintained now https://github.com/paularmstrong/swig)
```

Comment:
Hmm, would it make sense if I spend a couple hours getting rid of Swig? I had an old branch where I tested it (very briefly) but since it wasn't very high on priorities, I dropped it. Just published it here:

https://github.com/Trustroots/trustroots/tree/replace-swig-with-nunjucks